utils/pstor/src/etcd.rs

```diff
@@ -256,6 +261,42 @@ impl StoreKv for Etcd {
         Ok(result)
     }
 
+    /// Returns a vector of tuples. Each tuple represents a key-value pair. It paginates through all
+    /// the values for the prefix with limit.
+    async fn get_values_paged_all(
+        &mut self,
+        key_prefix: &str,
+        limit: i64,
+    ) -> Result<Vec<(String, Value)>, Error> {
+        let range_end = get_prefix_range_end(key_prefix).map_err(|_| Error::RangeEnd {
+            start_key: key_prefix.to_string(),
+        })?;
+
+        let mut prefix = key_prefix;
+        let mut first = true;
+        let mut all_values = vec![];
+        let mut values;
+
+        loop {
+            values = self.get_values_paged(prefix, limit, &range_end).await?;
+
+            if !first && values.get(0).is_some() {
+                values.remove(0);
+            }
+
+            first = false;
+
+            all_values.extend(values.clone());
+
+            if let Some((key, _)) = values.last() {
+                prefix = key;
+            } else {
+                break;
+            }
```

Comment:
```suggestion
            last = values.last().cloned();
            all_values.extend(values);

            if let Some((key, _)) = last() {
                prefix = key;
            } else {
                break;
            }
```

Reply:
Key becomes string with this, causes lifetime issue?